src/lib/parameterize.ts

```diff
@@ -68,6 +70,25 @@ const Parameterize = () => {
     parametersStorage = parameters;
   };
 
+  /**
+   * Creates a handler for managing unique unit names.
+   */
+  const setUniqueUnitName = (() => {
```

Comment:
we already have something like this in our codebase memoizedLog

you can reuse that with minor modifications to avoiding code duplication